.doks/doks-cluster/cluster.tf

```diff
@@ -5,43 +5,23 @@ resource "digitalocean_kubernetes_cluster" "codacy_k8s" {
 
   node_pool {
     name       = "codacy-doks-pool-${terraform.workspace}"
-    size       = var.node_type
-    node_count = var.num_nodes
+    size       = "s-2vcpu-2gb"
+    auto_scale = true
+    node_count = 1
+    min_nodes = 0
+    max_nodes = 0
   }
 
   provisioner "local-exec" {
     command = "doctl kubernetes cluster kubeconfig save ${var.cluster_name} --set-current-context"
   }
 }
 
-resource "kubernetes_namespace" "codacy" {
-  count = var.cluster_only ? 0 : 1
-  metadata {
-    name = var.main_namespace
-  }
-}
-
-resource "kubernetes_secret" "docker_credentials" {
-  depends_on = [kubernetes_namespace.codacy]
-  count = var.cluster_only ? 0 : 1
-  metadata {
-    name = "docker-credentials"
-    namespace = var.main_namespace
-  }
-  data = {
-    ".dockerconfigjson" = "{\"auths\": {\"https://index.docker.io/v1/\": {\"auth\": \"${base64encode("${var.docker_username}:${var.docker_password}")}\"}}}"
-  }
-
-  type = "kubernetes.io/dockerconfigjson"
-}
-
-resource "kubernetes_secret" "do_token" {
-  count = var.cluster_only ? 0 : 1
-  metadata {
-    name = "digitalocean"
-    namespace = "kube-system"
-  }
-  data = {
-    "access-token" = var.digital_ocean_token
-  }
+resource "digitalocean_kubernetes_node_pool" "auto-scale-pool-01" {
```

Comment:
This is the actual node pool that we use.
Due to shortcomings on the kubernetes terraform resource from digital ocean, if we want to change the node_type on this  pool, we will have to do so in multiple steps.
Changing the node_type for the environments on the config.yaml file will cause this node pool to be recreated (i.e. destroyed and created), effectively crushing to bits anything inside it. :'( 
What we can do is add a new node pool with the desired specification, then scale down this node pool. After all components have been moved to the new nodes, we can then scale up this node pool with the desired spec, and finally get rid of the second temporary node pool.
These were my findings, although this should not happen very often. I just want to raise my concern about this point, and that I am actually not happy with that approach, especially given that our terraform apply command is --auto-approve.
I am open to suggestions on how to actually prevent accidents from happening when changing the node_type.